### ISSUE TITLE
bind method is untyped

### ISSUE THREAD

**Issue by wanglam** [facebook/flow#1176]
There's an problem occurring in my project , which is save a context in my callback function.
The code is just like this: 
```javascript
/* @flow */
var do_something_async = function(cb:(x:number)=>void){
	setTimeout(cb(12),2000);
}
do_something_async(function(d){
	console.log(d.a);
}.bind({}));
```
The "d" is number , it doesn't have a property named "a" , so there should throw a error.Unfortunately, it will pass the test;
I find the Function declare code in flow ("flow/lib/core.js") 
```javascript
bind: Function$Prototype$Bind; // (thisArg: any, ...argArray: Array<any>) => any;
```
It seems like after calling bind method , it  will return any , so the test will pass.My temp temporary solution is to modify code like this:
```javascript
do_something_async((function(){
	return function(d){
		console.log(d.a);
	}
})());
```
Function.bind is a useful method , my solution is poor. So,how to fix the bind method?

**Comment by mroch** [comment]
the `Function$Prototype$Bind` stuff isn't released yet (will be in 0.20) and should fix your problem. are you trying with 0.19.1 or building from source?

**Comment by wanglam** [comment]
@mroch I have updated my flow version to 0.19.1 . This problem still exists.

**Comment by mroch** [comment]
ok. i think it'll be fixed in 0.20. (811f19a9fa39c7b3aa4bd470c66beaa7673b5b40)
